### ISSUE TITLE
WHOLE-INDEX not detected

### ISSUE THREAD

**Issue by wiba-cac** [Riverside-Software/sonar-openedge#866]
Hi,
I'm usint sonarlint in 12.2 but the WHOLE-INDEX is not shown (the other rules are ok, unused vars, "\", ..).
The XREF.XML files are created and include the "WHOLE-INDEX" node in the XML.

Are there any extra settings that have to be done for WHOLE-INDEX ?

The logs mention:
```
Parsing XML XREF file C:\temp\sonarLintXref\com\art\maglogcp.xref.xml
ActiveRule - Internal key null - Repository progress-rules - Rule com.progress.rules.OutdatedDigest
ActiveRule - Internal key null - Repository progress-rules - Rule com.progress.rules.WholeIndexAnalyzer
ActiveRule - Internal key null - Repository progress-rules - Rule com.progress.rules.BackslashInString
Initializing metadata of file file:/C:/pscworkspaces/wiba/tfs/FinErpWS12/com/art/maglogcp.p
ActiveRule - Internal key null - Repository progress-rules - Rule com.progress.rules.UnusedVariable
ActiveRule - Internal key null - Repository progress-rules - Rule com.progress.rules.WhereCanDo
1 files proparse'd, 1 XML files, 0 listing files, 0 failure(s), 0 NCLOCs
```

maglogcp.xref.xml
```
<Reference Reference-type="SEARCH" Object-identifier="iris.dfak">
      <Source-guid>yhq6T4WvU755FA8XIK+b5g</Source-guid>
      <File-num>1</File-num>
      <Ref-seq>439</Ref-seq>
      <Line-num>592</Line-num>
      <Object-context>XPKdfak</Object-context>
      <Access-mode/>
      <Data-member-ref/>
      <Temp-ref/>
      <Detail>WHOLE-INDEX</Detail>
      <Is-static>false</Is-static>
      <Is-abstract>false</Is-abstract>
    </Reference>
```

**Comment by gquerret** [comment]
First, make sure you're using version [5.5.1507](https://github.com/Riverside-Software/sonar-openedge/releases/download/V2.11.1/org.sonarlint.eclipse.site-5.5.1507.zip)

Then enable verbose output and analysis log in the console view:

![image](https://user-images.githubusercontent.com/6498723/106145882-54b33d00-6176-11eb-9adf-2c837b7beb4c.png)

Save your file again, and verify that the log output contains similar lines:

* XML XREF is activated
```
useXrefXML: true
XREFXMLPath: C:\Users\gqu\workspace\Test\xref
```

* Schema is loaded
```
Project has DB connections, looking for file C:\Users\gqu\workspace\Test\.sonarlint\dblist.txt
Reading dblist.txt
```

* XML XREF is loaded
```
Parsing XML XREF file C:\Users\gqu\workspace\Test\xref\test.xref.xml
```

THis simple sports2000 query will be highlighted:
![image](https://user-images.githubusercontent.com/6498723/106146439-f3d83480-6176-11eb-9e99-560fed71e1fd.png)

If you've manually defined sonar.oe.databases and aliases, make sure that the `iris` alias is known by SonarLint.

In case you're working on a very simple test, note that queries without `where` clause do not raise Whole-Index issues.

**Comment by wiba-cac** [comment]
Hi,
Thank you for the quick answer !

I think the version will be the problem. I use the standard version with PDSOE 12.2 and the build in sonarlint version.

Starting SonarLint for Eclipse 4.2.1203
Starting standalone SonarLint engine 4.2.1203...
Plugins : OpenEdge 2.6.4 (openedge)

Is it easy to upgrade this standalone SonarLint to 5.5.1507 or higher and how ?

Thank you

logging (everything seems to be configured) :
useXrefXML: true
XREFXMLPath: C:\temp\sonarLintXref

Parsing c:\temp\iris.df with alias iris
SonarLint side, using serialized file
Parsing c:\temp\icfdb.df with alias icfdb
SonarLint side, using serialized file

Parsing XML XREF file C:\temp\sonarLintXref\com\art\maglogcp.xref.xml

**Comment by gquerret** [comment]
> Is it easy to upgrade this standalone SonarLint to 5.5.1507 or higher and how ?

Unless you're in a controlled environment where you can't modify files, yes, it's easy. Download the ZIP file, and install it:
* choose 'Help' -> 'Install new software...'
* click on 'Add...', then on the 'Archive...' button and select the ZIP file you just downloaded
* check the two boxes 'SonarLint for Eclipse' and 'Optional Configuration Helpers'
* click on 'Next >' and follow the instructions
* restart Progress Developer Studio

Your log output seems to be correct, what is your query ? Does it contain a where clause ? A use-index clause ?

**Comment by wiba-cac** [comment]
I downloaded "org.sonarlint.eclipse.site-5.6.0.25634.zip" from https://binaries.sonarsource.com/SonarLint-for-Eclipse/releases/
But it doesn't show 'Optional Configuration Helpers'

Where can I download the correct zip-file with this option ? Do I need it ?

The query contains a where clause, without "use-index" and should show "WHOLE-INDEX".
Compile listing is showing "WHOLE-INDEX" too. And it would be nice if this warning shows up.

**Comment by gquerret** [comment]
You have to use a modified version of SonarLint to support Developer Studio ; it can be downloaded here: https://github.com/Riverside-Software/sonar-openedge/releases/tag/V2.11.1
Uninstall the SonarSource version that you've downloaded first.

**Comment by wiba-cac** [comment]
ok, V2.11.1 and  5.5.1507 are confusing, but in a small test .p , I get the warning ! great

In my orignal .p file which is 800 lines and has about 20 procedures , I get this Java error (which didn't appear before and it doesn't show the unused vars etc anymore)

Error during rule execution for [uri=file:/C:/pscworkspaces/wiba/tfs/FinErpWS12/com/art/maglogcp.p]
java.lang.UnsupportedOperationException: hasRelativePath
	at org.sonarsource.sonarlint.core.container.analysis.filesystem.DefaultFilePredicates.hasRelativePath(DefaultFilePredicates.java:67)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:284)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:297)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:90)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:142)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:183)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:172)
	at com.progress.rules.UnusedVariable.a(Unknown Source)
	at com.progress.rules.UnusedVariable.execute(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.sensorExecute(OpenEdgeProparseCheck.java:57)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:429)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:198)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.executeSensor(SensorsExecutor.java:80)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.execute(SensorsExecutor.java:71)
	at org.sonarsource.sonarlint.core.container.analysis.AnalysisContainer.doAfterStart(AnalysisContainer.java:132)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.startComponents(ComponentContainer.java:126)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.execute(ComponentContainer.java:111)
	at org.sonarsource.sonarlint.core.container.standalone.StandaloneGlobalContainer.analyze(StandaloneGlobalContainer.java:153)
	at org.sonarsource.sonarlint.core.StandaloneSonarLintEngineImpl.analyze(StandaloneSonarLintEngineImpl.java:90)
	at org.sonarlint.eclipse.core.internal.engine.StandaloneEngineFacade.lambda$1(StandaloneEngineFacade.java:99)
	at org.sonarlint.eclipse.core.internal.engine.StandaloneEngineFacade.withEngine(StandaloneEngineFacade.java:90)
	at org.sonarlint.eclipse.core.internal.engine.StandaloneEngineFacade.runAnalysis(StandaloneEngineFacade.java:98)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeStandaloneProjectJob.runAnalysis(AnalyzeStandaloneProjectJob.java:63)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeStandaloneProjectJob.runAnalysis(AnalyzeStandaloneProjectJob.java:1)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.run(AbstractAnalyzeProjectJob.java:389)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.runAnalysisAndUpdateMarkers(AbstractAnalyzeProjectJob.java:199)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.doRun(AbstractAnalyzeProjectJob.java:161)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob.run(AbstractSonarProjectJob.java:45)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
[maglogcp.p.txt](https://github.com/Riverside-Software/sonar-openedge/files/5888611/maglogcp.p.txt)

**Comment by gquerret** [comment]
This UnsupportedOperationException is definitely not good. Do you use linked folders in your project ?

**Comment by wiba-cac** [comment]
No, we don't use linked folders. 

All sources are in 1 dir with a lot of subdirs and some entries in PROPATH

the probleem is in the line : {checkerr.i &define-only = YES}

removing it (and all other code) makes the error disappear, but I don't get the WHOLE-INDEX warning (version is "Starting SonarLint for Eclipse 5.5.1507").

attached all the source code and df file (renamed to .txt files)

[checkerr.i.txt](https://github.com/Riverside-Software/sonar-openedge/files/5892554/checkerr.i.txt)
[maglogcp.p.txt](https://github.com/Riverside-Software/sonar-openedge/files/5892688/maglogcp.p.txt)
[iris.df.txt](https://github.com/Riverside-Software/sonar-openedge/files/5892699/iris.df.txt)

**Comment by gquerret** [comment]
I see errors when parsing the database dump file, that may be the root cause. I'll keep you posted.

**Comment by dominickdm** [comment]
I'm not sure if our issue is related, but we get the same error on several source files with the latest sonarlint eclipse plugin (5.5.1507). 
The 'ActiveRule' is not always the same..

Some examples:
ActiveRule - Internal key  - Repository rssw-oe-main - Rule eu.rssw.antlr.proparse.checks.NestedComments
Error during rule execution for [uri=file:/C:/Workspace/glims_dev_pro/glims-abl-sources/man/ord_q2i.p]
java.lang.UnsupportedOperationException: hasRelativePath
	at org.sonarsource.sonarlint.core.container.analysis.filesystem.DefaultFilePredicates.hasRelativePath(DefaultFilePredicates.java:67)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:284)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:297)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:90)
	at eu.rssw.antlr.proparse.checks.NestedComments.a(Unknown Source)
	at eu.rssw.antlr.proparse.checks.NestedComments$NestedCommentsCallback.visitNode(Unknown Source)
	at org.prorefactor.core.JPNode.walk(JPNode.java:785)
	at org.prorefactor.core.JPNode.walk(JPNode.java:787)
	at eu.rssw.antlr.proparse.checks.NestedComments.execute(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.sensorExecute(OpenEdgeProparseCheck.java:57)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:429)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:198)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.executeSensor(SensorsExecutor.java:80)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.execute(SensorsExecutor.java:71)
	at org.sonarsource.sonarlint.core.container.analysis.AnalysisContainer.doAfterStart(AnalysisContainer.java:132)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.startComponents(ComponentContainer.java:126)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.execute(ComponentContainer.java:111)
	at org.sonarsource.sonarlint.core.container.storage.StorageAnalyzer.analyze(StorageAnalyzer.java:75)
	at org.sonarsource.sonarlint.core.container.storage.StorageContainerHandler.analyze(StorageContainerHandler.java:85)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.lambda$analyze$0(ConnectedSonarLintEngineImpl.java:152)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:349)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:339)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.analyze(ConnectedSonarLintEngineImpl.java:149)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.lambda$13(ConnectedEngineFacade.java:354)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.withEngine(ConnectedEngineFacade.java:147)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.runAnalysis(ConnectedEngineFacade.java:353)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:70)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:1)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.run(AbstractAnalyzeProjectJob.java:389)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.runAnalysisAndUpdateMarkers(AbstractAnalyzeProjectJob.java:199)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.doRun(AbstractAnalyzeProjectJob.java:161)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob.run(AbstractSonarProjectJob.java:45)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)


ActiveRule - Internal key  - Repository rssw-oe-main - Rule eu.rssw.antlr.proparse.checks.BufferUsage
Error during rule execution for [uri=file:/C:/Workspace/glims_dev_pro/glims-abl-sources/man/ord_q2.p]
java.lang.UnsupportedOperationException: hasRelativePath
	at org.sonarsource.sonarlint.core.container.analysis.filesystem.DefaultFilePredicates.hasRelativePath(DefaultFilePredicates.java:67)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:284)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:297)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:90)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:142)
	at org.sonar.plugins.openedge.AbstractCommercialRule.createIssue(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:183)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:172)
	at eu.rssw.antlr.proparse.checks.BufferUsage.execute(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.sensorExecute(OpenEdgeProparseCheck.java:57)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:429)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:198)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.executeSensor(SensorsExecutor.java:80)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.execute(SensorsExecutor.java:71)
	at org.sonarsource.sonarlint.core.container.analysis.AnalysisContainer.doAfterStart(AnalysisContainer.java:132)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.startComponents(ComponentContainer.java:126)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.execute(ComponentContainer.java:111)
	at org.sonarsource.sonarlint.core.container.storage.StorageAnalyzer.analyze(StorageAnalyzer.java:75)
	at org.sonarsource.sonarlint.core.container.storage.StorageContainerHandler.analyze(StorageContainerHandler.java:85)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.lambda$analyze$0(ConnectedSonarLintEngineImpl.java:152)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:349)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:339)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.analyze(ConnectedSonarLintEngineImpl.java:149)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.lambda$13(ConnectedEngineFacade.java:354)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.withEngine(ConnectedEngineFacade.java:147)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.runAnalysis(ConnectedEngineFacade.java:353)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:70)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:1)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.run(AbstractAnalyzeProjectJob.java:389)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.runAnalysisAndUpdateMarkers(AbstractAnalyzeProjectJob.java:199)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.doRun(AbstractAnalyzeProjectJob.java:161)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob.run(AbstractSonarProjectJob.java:45)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)


ActiveRule - Internal key  - Repository rssw-oe-main - Rule eu.rssw.antlr.proparse.checks.EmptyBlock
Error during rule execution for [uri=file:/C:/Workspace/glims_dev_pro/glims-abl-sources/man/bd_ord2.p]
java.lang.UnsupportedOperationException: hasRelativePath
	at org.sonarsource.sonarlint.core.container.analysis.filesystem.DefaultFilePredicates.hasRelativePath(DefaultFilePredicates.java:67)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:284)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:297)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:90)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:142)
	at org.sonar.plugins.openedge.AbstractCommercialRule.createIssue(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:183)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:172)
	at eu.rssw.antlr.proparse.checks.EmptyBlock.execute(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.sensorExecute(OpenEdgeProparseCheck.java:57)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:429)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:198)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.executeSensor(SensorsExecutor.java:80)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.execute(SensorsExecutor.java:71)
	at org.sonarsource.sonarlint.core.container.analysis.AnalysisContainer.doAfterStart(AnalysisContainer.java:132)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.startComponents(ComponentContainer.java:126)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.execute(ComponentContainer.java:111)
	at org.sonarsource.sonarlint.core.container.storage.StorageAnalyzer.analyze(StorageAnalyzer.java:75)
	at org.sonarsource.sonarlint.core.container.storage.StorageContainerHandler.analyze(StorageContainerHandler.java:85)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.lambda$analyze$0(ConnectedSonarLintEngineImpl.java:152)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:349)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:339)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.analyze(ConnectedSonarLintEngineImpl.java:149)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.lambda$13(ConnectedEngineFacade.java:354)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.withEngine(ConnectedEngineFacade.java:147)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.runAnalysis(ConnectedEngineFacade.java:353)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:70)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:1)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.run(AbstractAnalyzeProjectJob.java:389)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.runAnalysisAndUpdateMarkers(AbstractAnalyzeProjectJob.java:199)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.doRun(AbstractAnalyzeProjectJob.java:161)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob.run(AbstractSonarProjectJob.java:45)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)


ActiveRule - Internal key  - Repository rssw-oe-main - Rule eu.rssw.antlr.proparse.checks.VariableReadButNotAssigned
Error during rule execution for [uri=file:/C:/Workspace/glims_dev_pro/glims-abl-sources/man/bd_ord2.p]
java.lang.UnsupportedOperationException: hasRelativePath
	at org.sonarsource.sonarlint.core.container.analysis.filesystem.DefaultFilePredicates.hasRelativePath(DefaultFilePredicates.java:67)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:284)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.getInputFile(OpenEdgeProparseCheck.java:297)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:90)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.createIssue(OpenEdgeProparseCheck.java:142)
	at org.sonar.plugins.openedge.AbstractCommercialRule.createIssue(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:183)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.reportIssue(OpenEdgeProparseCheck.java:172)
	at eu.rssw.antlr.proparse.checks.VariableReadButNotAssigned.a(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1620)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at eu.rssw.antlr.proparse.checks.VariableReadButNotAssigned.a(Unknown Source)
	at eu.rssw.antlr.proparse.checks.VariableReadButNotAssigned.execute(Unknown Source)
	at org.sonar.plugins.openedge.api.checks.OpenEdgeProparseCheck.sensorExecute(OpenEdgeProparseCheck.java:57)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:429)
	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:198)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.executeSensor(SensorsExecutor.java:80)
	at org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.execute(SensorsExecutor.java:71)
	at org.sonarsource.sonarlint.core.container.analysis.AnalysisContainer.doAfterStart(AnalysisContainer.java:132)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.startComponents(ComponentContainer.java:126)
	at org.sonarsource.sonarlint.core.container.ComponentContainer.execute(ComponentContainer.java:111)
	at org.sonarsource.sonarlint.core.container.storage.StorageAnalyzer.analyze(StorageAnalyzer.java:75)
	at org.sonarsource.sonarlint.core.container.storage.StorageContainerHandler.analyze(StorageContainerHandler.java:85)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.lambda$analyze$0(ConnectedSonarLintEngineImpl.java:152)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:349)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.withReadLock(ConnectedSonarLintEngineImpl.java:339)
	at org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.analyze(ConnectedSonarLintEngineImpl.java:149)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.lambda$13(ConnectedEngineFacade.java:354)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.withEngine(ConnectedEngineFacade.java:147)
	at org.sonarlint.eclipse.core.internal.engine.connected.ConnectedEngineFacade.runAnalysis(ConnectedEngineFacade.java:353)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:70)
	at org.sonarlint.eclipse.core.internal.jobs.AnalyzeConnectedProjectJob.runAnalysis(AnalyzeConnectedProjectJob.java:1)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.run(AbstractAnalyzeProjectJob.java:389)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.runAnalysisAndUpdateMarkers(AbstractAnalyzeProjectJob.java:199)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractAnalyzeProjectJob.doRun(AbstractAnalyzeProjectJob.java:161)
	at org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob.run(AbstractSonarProjectJob.java:45)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

**Comment by gquerret** [comment]
Thank you @dominickdm for the additional stack traces !
Issue has been solved, that was coming from changes in the SonarLint API, unfortunately in a part where there are no automated tests... While the full regression suite is being executed, you can try this version: https://dl.rssw.eu/org.sonarlint.eclipse.site-5.5.1508.zip

**Comment by gquerret** [comment]
@dominickdm @wiba-cac PDSOE issue fixed in version 2.12.1 (and SonarLint 5.5.1509).

@wiba-cac Let me know if you still have an issue with whole-index

**Comment by wiba-cac** [comment]
Hi,
I installed 5.5.1509. The java stack-trace errors are gone, but I still have the issue with whole-index (not being shown).

I comment almost everything out. The code with the whole-index is left.

The XML file contains the WHOLE-INDEX issue ..

But the logs shows
1 files proparse'd, 1 XML files, 0 listing files, 0 failure(s), 0 NCLOCs
Found 0 issue(s)

it works fine in a small test.p ..

[maglogcp.txt](https://github.com/Riverside-Software/sonar-openedge/files/5902704/maglogcp.txt)
[logic.i.txt](https://github.com/Riverside-Software/sonar-openedge/files/5902805/logic.i.txt)

**Comment by gquerret** [comment]
As soon as code has `&ANALYZE_SUSPEND` declarations, the parser tries to identify AppBuilder managed code sections, so that issues are not reported there. In your case, the `writeEndTransValidate` procedure is incorrectly flagged as AppBuilder code section due to the `_DB-REQUIRED` attribute in `&ANALYZE_SUSPEND` (is that specific to Dynamics ?). That will be fixed in next version. Unfortunately, there's no way to work around the issue for now (apart from removing the `_DB-REQUIRED` attribute, but I don't think this is something you want to do...)

**Comment by wiba-cac** [comment]
i don't know if this is specific to Dynamics, because is Dynamics is gone in v12 Appbuilder and the toggle is still in the UIB screen ..
"builds" in Dynamics use this in order to check if this is client code or Appserver code.

So, we will wait for the next version, thanks a lot.


![image](https://user-images.githubusercontent.com/64423711/106473495-200bf200-64a4-11eb-9852-dcc63fc40af4.png)

**Comment by gquerret** [comment]
Issue has been fixed in its branch, that will be merged this week the main development line.
If you have time for tests, I can push a snapshot of the Eclipse plugin.

**Comment by wiba-cac** [comment]
I can test this tomorrow

**Comment by gquerret** [comment]
This snapshot is available here: https://dl.rssw.eu/org.sonarlint.eclipse.site-5.5.1508.zip

**Comment by gquerret** [comment]
This snapshot is available here: https://dl.rssw.eu/org.sonarlint.eclipse.site-5.5.1510.zip
